lib/Dialect/TritonGPU/Transforms/HoistTMEMAlloc.cpp

```diff
@@ -38,6 +38,105 @@ using TMEMTokenLoadOp = HasToken<ttng::TMEMLoadOp>;
 using TMEMTokenStoreOp = HasToken<ttng::TMEMStoreOp>;
 using TMEMTokenAllocOp = HasToken<ttng::TMEMAllocOp>;
 
+// Determine if a store operation is overridden by an MMA op that
+// will ignore the accumulator. Whenever the accumulator is ignored
+// the load should be predicated.
+bool isInitialStoreUnused(TMEMTokenAllocOp allocOp, Operation *loadOp,
+                          scf::ForOp parentFor) {
+  Value bufferValue = allocOp->getResults()[0];
+  // Look for exactly one store. If there are multiple stores, we can't
+  // match accurately. An MMA could replace a store, but the initial
```

Comment:
Could you help me understand the purpose of this check? I don't quite get what pattern we're looking for.

Reply:
This pattern mostly exists so I can prove the load and store are in the same basic block. I'm concerned with a pattern like this:
```
store
if (...) {
 # use the store here?
 ...
}
mma
load
```
This could still match depending on if there is user in the if, but its more complicated. Requiring 1 store is just a simplification because since this pass introduces TMEM store it shouldn't match. However, if these passes change it helps future proof the implementation.